superset-frontend/plugins/plugin-chart-echarts/src/Bubble/transformProps.ts

```diff
@@ -0,0 +1,207 @@
+}
+
+export default function transformProps(chartProps: EchartsBubbleChartProps) {
+  const { height, width, hooks, queriesData, formData, inContextMenu } =
+    chartProps;
+
+  const { data = [] } = queriesData[0];
+  const {
+    x,
+    y,
+    size,
+    entity,
+    maxBubbleSize,
+    colorScheme,
+    series: bubbleSeries,
+    xAxisLabel: bubbleXAxisTitle,
+    yAxisLabel: bubbleYAxisTitle,
+    xAxisFormat,
+    yAxisFormat,
+    yAxisBounds,
+    logXAxis,
+    logYAxis,
+    xAxisTitleMargin,
+    yAxisTitleMargin,
+    truncateYAxis,
+    xAxisLabelRotation,
+    yAxisLabelRotation,
+    tooltipSizeFormat,
+    opacity,
+  }: EchartsBubbleFormData = { ...DEFAULT_FORM_DATA, ...formData };
+
+  const colorFn = CategoricalColorNamespace.getScale(colorScheme as string);
+
+  const legends: string[] = [];
+  const series: ScatterSeriesOption[] = [];
+
+  const xAxisLabel: string = getMetricLabel(x);
+  const yAxisLabel: string = getMetricLabel(y);
+  const sizeLabel: string = getMetricLabel(size);
+
+  const refs: Refs = {};
+
+  data.forEach(datum => {
+    const name = (bubbleSeries ? datum[bubbleSeries] : datum[entity]) as string;
+    const bubbleSeriesValue = bubbleSeries ? datum[bubbleSeries] : null;
+
+    series.push({
+      name,
+      data: [
+        [
+          datum[xAxisLabel],
+          datum[yAxisLabel],
+          datum[sizeLabel],
+          datum[entity],
+          bubbleSeriesValue as any,
+        ],
+      ],
+      type: 'scatter',
+      itemStyle: { color: colorFn(name), opacity },
+    });
+    legends.push(name);
+  });
+
+  normalizeSymbolSize(series, maxBubbleSize);
+
+  const xAxisFormatter = getNumberFormatter(xAxisFormat);
+  const yAxisFormatter = getNumberFormatter(yAxisFormat);
+  const tooltipSizeFormatter = getNumberFormatter(tooltipSizeFormat);
+
+  const [min, max] = yAxisBounds.map(parseYAxisBound);
+
+  const echartOptions: EChartsCoreOption = {
+    series,
+    xAxis: {
+      axisLabel: { formatter: xAxisFormatter },
+      splitLine: {
+        lineStyle: {
+          type: 'dashed',
+        },
+      },
+      nameRotate: xAxisLabelRotation,
+      scale: true,
+      name: bubbleXAxisTitle,
+      nameLocation: 'middle',
+      nameTextStyle: {
+        fontWight: 'bolder',
+      },
+      nameGap: xAxisTitleMargin || 30,
+      type: logXAxis ? AxisType.log : AxisType.value,
+    },
+    yAxis: {
+      axisLabel: { formatter: yAxisFormatter },
+      splitLine: {
+        lineStyle: {
+          type: 'dashed',
+        },
+      },
+      nameRotate: yAxisLabelRotation,
+      scale: truncateYAxis,
+      name: bubbleYAxisTitle,
+      nameLocation: 'middle',
+      nameTextStyle: {
+        fontWight: 'bolder',
+      },
+      nameGap: yAxisTitleMargin || 50,
+      min,
+      max,
+      type: logYAxis ? AxisType.log : AxisType.value,
+    },
+    legend: {
+      ...getLegendProps(LegendType.Scroll, LegendOrientation.Top, true),
+      data: legends,
+    },
```

Comment:
Let's make the legend customizable like in the other charts, both so it can be disabled/enabled and also changeing the orientation etc (there should be controls and utils that can be readily reused)